contracts/L1ForkArbitrator.sol

```diff
@@ -0,0 +1,117 @@
+import {IForkingManager} from "./interfaces/IForkingManager.sol";
+import {IForkonomicToken} from "./interfaces/IForkonomicToken.sol";
+import {IForkableStructure} from "./interfaces/IForkableStructure.sol";
+
+contract L1ForkArbitrator is IArbitratorCore, IArbitratorErrors {
+    IForkingManager public forkmanager;
+    IRealityETH public realitio;
+    IForkonomicToken public token;
+
+    /// @notice Could not deduct fee
+    error CouldNotDeductFee();
+
+    /// @notice Not forked yet
+    error NotForkedYet();
+
+    /// @notice This arbitrator can only arbitrate one dispute in its lifetime
+    error CanOnlyArbitrateOneDispute();
+
+    /// @notice The question ID must be supplied
+    error MissingQuestionID();
+
+    bytes32 public arbitratingQuestionId;
+    address public payer;
+
+    constructor(address _realitio, address _forkmanager, address _token) {
+        realitio = IRealityETH(_realitio);
+        forkmanager = IForkingManager(_forkmanager);
+        token = IForkonomicToken(_token);
+    }
+
+    /* solhint-disable quotes */
+    string public metadata = '{"erc20": true}';
+
+    /// @notice Return the dispute fee for the specified question. 0 indicates that we won't arbitrate it.
+    /// @dev Uses a general default, but can be over-ridden on a question-by-question basis.
+    function getDisputeFee(bytes32) public view returns (uint256) {
+        return forkmanager.arbitrationFee();
+    }
+
+    /// @notice Request arbitration, freezing the question until we send submitAnswerByArbitrator
+    /// @dev Requires payment in token(), which the UI as of early 2024 will not handle.
+    /// @param _questionId The question in question
+    /// @param _maxPrevious If specified, reverts if a bond higher than this was submitted after you sent your transaction.
+    /// NB TODO: This is payable because the interface expects native tokens. Consider changing the interface or adding an ERC20 version.
+    function requestArbitration(
+        bytes32 _questionId,
+        uint256 _maxPrevious
+    ) external payable returns (bool) {
+        if (arbitratingQuestionId != bytes32(0))
+            revert CanOnlyArbitrateOneDispute();
+        if (_questionId == bytes32(0)) revert MissingQuestionID();
+
+        uint256 fee = getDisputeFee(_questionId);
+        if (fee == 0)
+            revert TheArbitratorMustHaveSetANonZeroFeeForTheQuestion();
+
+        // First we transfer the fee to ourselves.
+        if (!token.transferFrom(msg.sender, address(this), fee))
+            revert CouldNotDeductFee();
+
+        payer = msg.sender;
+        arbitratingQuestionId = _questionId;
+
+        realitio.notifyOfArbitrationRequest(
+            _questionId,
+            msg.sender,
+            _maxPrevious
+        );
+
+        // Now approve so that the fee can be transferred right out to the ForkingManager
+        if (!token.approve(address(forkmanager), fee))
+            revert CouldNotDeductFee();
```

Comment:
it seems the error message does not fit to the approve statement?



Why don't we approve once the uint.max during deployment?

Reply:
It only ever gets called once (since you'll have to call the child next time) and may get called zero times (if the system never forks). So it seems like a waste of gas to call it before we need it.

Reply:
Fixed the error message in [149c4c4](https://github.com/RealityETH/subjectivocracy/pull/236/commits/149c4c422143ce284ca508444f91882b3ea8b91b)

(Although it should probably not be able to fail...)